src/core/chuck_oo.cpp

```diff
@@ -632,8 +632,30 @@ t_CKINT Chuck_Array4::pop_back( )
     return 1;
 }
 
+//-----------------------------------------------------------------------------
+// name: pop_out()
+// desc: ...
+//-----------------------------------------------------------------------------
+t_CKINT Chuck_Array4::pop_out( t_CKUINT pos )
+{
+	// check
+	if ( m_vector.size() == 0 || pos<0 || pos>=m_vector.size())
+		return 0;
 
+	if( m_is_obj )
+	{
+		// get pointer
+		Chuck_Object * v = (Chuck_Object *)m_vector[m_vector.size()-1];
+		// if not null, release
+		if( v ) v->release();
+	}
```

Comment:
It's been a while since I touched arrays. Can you remind me why this one has special handling for the m_is_obj flag and why you'd want to release() here?

Reply:
I really just did it because pop_back did it for Chuck_Array4, so there's probably a reason for it. Also, thank you for pointing it out, I realized it wasn't freeing the right pointer! It was releasing the last pointer in the array (like you'd want with pop_back) instead of the pos'th pointer (like you'd want with pop_out), but I fixed that.